codegen/src/getable.rs

```diff
@@ -283,19 +283,27 @@ where
             .ident
             .as_ref()
             .expect("Struct fields always have names");
-
+        let quoted_field_ident = format!("{}", quote! { #field_ident });
         quote! {
-            #field_ident: if let Some(val) = data.get_variant(#idx) {
+            #field_ident: if let Some(val) = inner_data.lookup_field(vm, #quoted_field_ident) {
                 <#field_ty as _gluon_api::Getable<'__vm, '__value>>::from_value(vm, val)
             } else {
                 panic!("Enum does not contain data at index '{}'. Do the type definitions match?", #idx)
```

Comment:
Just a nitpick, but this error message requires some rewording. That would also remove the need for the extra `enumerate` as far as I can tell.